tests/test_web_functional.py

```diff
@@ -1663,20 +1668,50 @@ def make_handler(appname):
         on_connection_create_start)
     trace_config.on_connection_create_end.append(
         on_connection_create_end)
+    trace_config.on_dns_resolvehost_start.append(
+        on_dns_resolvehost_start)
+    trace_config.on_dns_resolvehost_end.append(
+        on_dns_resolvehost_end)
 
     app = web.Application()
     app.router.add_get('/redirector', redirector)
     app.router.add_get('/redirected', redirected)
+    server = await aiohttp_server(app)
+
+    class FakeResolver:
+        _LOCAL_HOST = {0: '127.0.0.1',
+                       socket.AF_INET: '127.0.0.1'}
+
+        def __init__(self, fakes):
+            """fakes -- dns -> port dict"""
+            self._fakes = fakes
+            self._resolver = aiohttp.DefaultResolver()
+
+        async def resolve(self, host, port=0, family=socket.AF_INET):
+            fake_port = self._fakes.get(host)
+            if fake_port is not None:
+                return [{'hostname': host,
+                         'host': self._LOCAL_HOST[family], 'port': fake_port,
+                         'family': socket.AF_INET, 'proto': 0,
+                         'flags': socket.AI_NUMERICHOST}]
+            else:
+                return await self._resolver.resolve(host, port, family)
 
-    client = await aiohttp_client(app, trace_configs=[trace_config])
+    resolver = FakeResolver({'example.com': server.port})
+    connector = aiohttp.TCPConnector(resolver=resolver)
+    client = aiohttp.ClientSession(connector=connector,
+                                   trace_configs=[trace_config])
 
-    await client.get('/redirector', data="foo")
+    await client.get('http://example.com/redirector', data="foo")
```

Comment:
a `f"localhost:{server.port}/redirector"` does not fire a DNS event? If it does we wouldn't need it the `FakeResolver`

Reply:
Resolver doesn't perform DNS lookup for IP addresses, only for domain names.
`localhost` should be sufficient but unfortunately there are too many boxes with declared but actually disable IPv6 support.

Reply:
sorry I don't get the following sentence.

> there are too many boxes with declared but actually disabled IPv6 support.

Reply:
I dont want to block the PR just for a test that in any case can be at any moment refactorized.

Reply:
Ok, I'll try to describe. Some broken environments (Travis-CI included IIRC) resolves `localhost` to both `127.0.0.1` and `::1` but IPv6 stack is disabled.
By this communication over IPv6 `::1` fails.
That's why `aiohttp.test_utils` and aiohttp test suite use IPv4 `127.0.0.1` explicitly.
The situation could be improved I hope but working on it is definitely out if the issue scope.